data_steward/test/unit_test/data_steward/cdr_cleaner/cleaning_rules/drop_participants_without_ppi_or_ehr_test.py

```diff
@@ -0,0 +1,123 @@
+# Python imports
+import random
+import string
+import unittest
+
+# Third party imports
+from googleapiclient.errors import HttpError
+from jinja2 import Template
+
+# Project imports
+import bq_utils
+import common
+from constants import bq_utils as bq_consts
+from cdr_cleaner import clean_cdr_engine
+from constants.cdr_cleaner import clean_cdr as clean_consts
+from cdr_cleaner.cleaning_rules import drop_rows_for_missing_persons
+from cdr_cleaner.cleaning_rules import drop_participants_without_ppi_or_ehr
+import resources
+from test.unit_test import test_util
+
+
+# Participant 1: no data (to be removed)
+# Participant 2: has the basics only
+# Participant 3: has EHR data only
+# Participant 4: has both basics and EHR
+# Participant 5: has only RDR consent (to be removed)
+# Participant 6: has EHR observation only
+INSERT_FAKE_PARTICIPANTS_TMPLS = [
+    # XXX: Ideally these tests should not manipulate concept table, not currently hermetic.
```

Comment:
@mark-velez  Do we have a way to ensure that the concept/concept_ancestor tables exist in the environment under testing? Do we have a better way to inject test data (here I used a forced nullable approach to make test data insertion easier)?

Reply:
@calbach we do not have a clear way to ensure the vocabulary tables exist in the test environment. It's not the best solution, but we typically load the vocabulary tables once in all our test datasets (e.g. using `table_copy.sh`). Thereafter we do not touch the vocabulary tables. The function `test_util.delete_all_tables` which is often used in our `tearDown`s does **not** delete the vocabulary tables.

Reply:
Is there a way to invoke this setup from an integration test, or do the integration tests generally assume that these tables have already been created? (and will this be true for someone who is running Circle for their own PR?).

Reply:
The integration tests generally and unfortunately assume that these tables have already been created. One possible solution might be to add a step in an init script which loads the vocabulary tables if they do not already exist. A more hermetic solution though might need to ensure the tables are untampered with (schema is correct, rows are as expected). Another idea might be to create a view for each vocabulary table onto some shared (somehow trusted, maybe immutable?) dataset which contains the actual vocabulary tables.